flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/ContextResolvedTableJsonDeserializer.java

```diff
@@ -172,9 +172,10 @@ private boolean isLookupEnabled(CatalogPlanRestore planRestoreOption) {
     static ValidationException missingIdentifier() {
         return new ValidationException(
                 String.format(
-                        "The table cannot be deserialized, as no identifier is present within the JSON, "
-                                + "but lookup is forced by '%s' == '%s'. "
-                                + "Either allow restoring table from the catalog with '%s' == '%s' | '%s' or make sure you don't use anonymous tables when generating the plan.",
+                        "The table cannot be deserialized as no identifier is present in the persisted plan."
```

Comment:
Unless I'm missing something, those cases (of exceptions) are not covered?
Shouldn't we have tests for all those error messages?

Reply:
I agree. I will put this on my list for polishing the table serialization.